CustomerContactBook/Services/CustomersService.cs

```diff
@@ -0,0 +1,122 @@
+﻿using CustomerContactBook.Database;
+using CustomerContactBook.Database.Tables;
+using CustomerContactBook.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace CustomerContactBook.Services
+{
+    public class CustomersService
+    {
+        private readonly ContactBookContext _context;
+
+        public CustomersService(ContactBookContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<CustomerModel>> GetCustomers()
+        {
+            var result =  await _context.Customers.ToListAsync();
+            return result.Select(toCustomerModel).ToList();
+        }
+
+        public async Task<CustomerModel> GetCustomer(long id)
+        {
+            var customer = await _context.Customers.FindAsync(id);
+
+            if (customer == null)
+            {
+                return null;
+            }
+
+            return toCustomerModel(customer);
+        }
+
+        public async Task<bool?> PutCustomer(long id, CustomerModel model)
+        {
+            var customer = toCustomer(model);
+            if (customer == null)
+            {
+                return null;
+            }
+            _context.Entry(customer).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException) when (!CustomerExists(id))
+            {
+                return null;
+            }
+
+            return true;
+        }
+
+        public async Task<CustomerModel> PostCustomer(CustomerModel model)
+        {
+            var customer = new Customer
+            {
+                Id = model.Id,
```

Comment:
This is a fully working way to set an id, but it's always better to let DB takes care of it.